src/quo2/components/list_items/saved_address/view.cljs

```diff
@@ -0,0 +1,69 @@
+(ns quo2.components.list-items.saved-address.view
+  (:require [quo2.components.avatars.wallet-user-avatar :as wallet-user-avatar]
+            [quo2.components.markdown.text :as text]
+            [quo2.foundations.colors :as colors]
+            [quo2.theme :as quo.theme]
+            [react-native.core :as rn]
+            [quo2.components.list-items.saved-address.style :as style]
+            [reagent.core :as reagent]
+            [quo2.components.icon :as icon]
+            [clojure.string :as string]))
+
+(defn- internal-view
+  []
+  (let [state       (reagent/atom :default)
+        active?     (atom false)
+        timer       (atom nil)
+        on-press-in (fn []
+                      (when-not (= @state :selected)
+                        (reset! timer (js/setTimeout #(reset! state :pressed) 100))))]
+    (fn [{:keys [blur? user-props customization-color
+                 on-press
+                 on-options-press
+                 theme]
+          :or   {customization-color :blue
+                 blur?               false}}]
+      (let [names        (string/split (:name user-props) " ")
+            first-name   (first names)
+            last-name    (last names)
+            on-press-out (fn []
+                           (let [new-state (if @active? :default :active)]
+                             (when @timer (js/clearTimeout @timer))
+                             (reset! timer nil)
+                             (reset! active? (= new-state :active))
+                             (reset! state new-state)
+                             (when on-press
+                               (on-press))))]
+        [rn/pressable
+         {:style               (style/container
+                                {:state @state :blur? blur? :customization-color customization-color})
+          :on-press-in         on-press-in
+          :on-press-out        on-press-out
+          :accessibility-label :container}
+         [rn/view {:style style/left-container}
```

Comment:
Just a suggestion to break the tree to reduce the re-rendering overhead.

The difference here is that Reagent won't re-compute `left-container` when `state`, `timer`, `active?`, `on-press`, and `on-options-press` are changed.

```diff
diff --git a/src/quo2/components/list_items/saved_address/view.cljs b/src/quo2/components/list_items/saved_address/view.cljs
index ace513201..e21d19d02 100644
--- a/src/quo2/components/list_items/saved_address/view.cljs
+++ b/src/quo2/components/list_items/saved_address/view.cljs
@@ -9,6 +9,29 @@
             [quo2.components.icon :as icon]
             [clojure.string :as string]))
 
+(defn left-container
+  [{:keys [blur? theme name address customization-color]}]
+  (let [names      (string/split name " ")
+        first-name (first names)
+        last-name  (last names)]
+    [rn/view {:style style/left-container}
+     [wallet-user-avatar/wallet-user-avatar
+      {:size   :medium
+       :f-name first-name
+       :l-name last-name
+       :color  customization-color}]
+     [rn/view {:style style/account-container}
+      [text/text
+       {:weight :semi-bold
+        :size   :paragraph-1}
+       name]
+      [text/text {:size :paragraph-2}
+       [text/text
+        {:size   :paragraph-2
+         :weight :monospace
+         :style  (style/account-address blur? theme)}
+        address]]]]))
+
 (defn- internal-view
   []
   (let [state       (reagent/atom :default)
@@ -23,10 +46,7 @@
                  theme]
           :or   {customization-color :blue
                  blur?               false}}]
-      (let [names        (string/split (:name user-props) " ")
-            first-name   (first names)
-            last-name    (last names)
-            on-press-out (fn []
+      (let [on-press-out (fn []
                            (let [new-state (if @active? :default :active)]
                              (when @timer (js/clearTimeout @timer))
                              (reset! timer nil)
@@ -40,23 +60,12 @@
           :on-press-in         on-press-in
           :on-press-out        on-press-out
           :accessibility-label :container}
-         [rn/view {:style style/left-container}
-          [wallet-user-avatar/wallet-user-avatar
-           {:size   :medium
-            :f-name first-name
-            :l-name last-name
-            :color  (:customization-color user-props)}]
-          [rn/view {:style style/account-container}
-           [text/text
-            {:weight :semi-bold
-             :size   :paragraph-1}
-            (:name user-props)]
-           [text/text {:size :paragraph-2}
-            [text/text
-             {:size   :paragraph-2
-              :weight :monospace
-              :style  (style/account-address blur? theme)}
-             (:address user-props)]]]]
+         [left-container
+          {:blur?               blur?
+           :theme               theme
+           :name                (:name user-props)
+           :address             (:address user-props)
+           :customization-color (:customization-color user-props)}]
          [rn/pressable
           {:accessibility-label :options-button
            :on-press            #(when on-options-press

```

Reply:
Agree, updated implementation